### ISSUE TITLE
Map file format does not use standard signed LEB128 as specified

### ISSUE THREAD

**Issue by wentam** [mapsforge/mapsforge#1156]
In the Mapsforge map format spec [here](https://github.com/mapsforge/mapsforge/blob/master/docs/Specification-Binary-Map-File.md), VBE-S is specified as being signed LEB128. It's not.

While the Mapsforge format is similar, it differs for negative numbers:
- Signed LEB128 uses two's complement for negative value storage. 
- Mapsforge format is using sign-and-magnitude.

Two's complement is the most common format used to store negative numbers in computing. Most programming languages store negative values this way.

**For example, to encode -10995:**

What LEB128 specifies:                                                                                   
```
111111101010100001101      - -10995 big endian two's complement
1111111 0101010 0001101    - make groups of 7                  
01111111 10101010 10001101 - add continuation bits             
10001101 10101010 01111111 - use little-endian byte order      
```
                                                                     
What the Mapsforge encoder does:                                                                                                                     
```
000000010101011110011      - positive 10995 big endian    
0000000 1010101 1110011    - make groups of 7             
00000000 11010101 11110011 - add continuation bits
11110011 11010101 00000000 - use little-endian byte order
11110011 11010101 01000000 - flip sign bit
```                           

If the format is not to be corrected the specification should at least be changed to reflect the true format. I would probably state something along the lines of "VBE-S is a similar encoding to [LEB128](https://en.wikipedia.org/wiki/LEB128), but uses sign-and-magnitude as opposed to two's complement for negative values" before going into the details.

The Mapsforge specification does go into further details, but nothing that would differentiate it from LEB128's two's complement format.

This costed me more time than I'd like to admit thinking I wrote my signed LEB128 encoder wrong when it was actually the Mapsforge format differing from the spec.

**Comment by devemux86** [comment]
Those descriptions are not from Mapsforge authors, but contributed by community via https://github.com/mapsforge/mapsforge/commit/02ed5176b5f41529ba368d2dd140650cd04fadc0.

We can revert documentation back to normal.
And can always study map reader to see how to read Mapsforge map files.

**Comment by wentam** [comment]
I see!

I would still recommend clearly specifying that the format is sign-and-magnitude in that document, as one can only assume it uses the usual two's complement format without an alternative being specified.

**Comment by devemux86** [comment]
Can you provide a pull request with the doc changes?

**Comment by devemux86** [comment]
Fixed by #1157.
